base/applications/mscutils/eventvwr/eventvwr.c

```diff
@@ -4007,11 +4007,79 @@ InitPropertiesDlg(HWND hDlg, PEVENTLOG EventLog)
     }
 }
 
+static
+VOID
+SavePropertiesDlg(HWND hDlg, PEVENTLOG EventLog)
+{
+    LPWSTR lpLogName = EventLog->LogName;
+
+    DWORD dwType;
+    DWORD dwMaxSize = 0, dwRetention = 0;
+    HKEY hLogKey;
+    WCHAR *KeyPath;    
+    SIZE_T cbKeyPath;
+
+    if (EventLog->Permanent)
+    {
+
+    cbKeyPath = (wcslen(EVENTLOG_BASE_KEY) + wcslen(lpLogName) + 1) * sizeof(WCHAR);
+    KeyPath = HeapAlloc(GetProcessHeap(), 0, cbKeyPath);
+    if (!KeyPath)
+    {
+        ShowWin32Error(ERROR_NOT_ENOUGH_MEMORY);
+        return;
+    }
+
+    StringCbCopyW(KeyPath, cbKeyPath, EVENTLOG_BASE_KEY);
+    StringCbCatW(KeyPath, cbKeyPath, lpLogName);
+
+    /* Try to create/open the Event Viewer user key */
+    if (RegCreateKeyExW(HKEY_LOCAL_MACHINE,
+                        KeyPath,
+                        0,
+                        NULL,
+                        REG_OPTION_NON_VOLATILE,
+                        KEY_READ | KEY_WRITE,
+                        NULL,
+                        &hLogKey,
+                        NULL) != ERROR_SUCCESS)
+    {
+        ShowWin32Error(GetLastError());
+		return;
+    }
+   
+    HeapFree(GetProcessHeap(), 0, KeyPath);
+
+    dwMaxSize = GetDlgItemInt(hDlg, IDC_EDIT_MAXLOGSIZE, NULL, FALSE)*1024;
+    dwType = REG_DWORD;
+    RegSetValueExW(hLogKey,
+                   L"MaxSize",
+                   0,
+                   dwType,
+                   (LPBYTE)&dwMaxSize,
+                   sizeof(dwMaxSize));    
+
+    dwRetention = GetDlgItemInt(hDlg, IDC_EDIT_EVENTS_AGE, NULL, FALSE)*(24*3600);
+    dwType = REG_DWORD;
```

Comment:
There is no need to assign your variable again with `REG_DWORD` if no other code alters its data.

Reply:
ok